tsl/src/bgw_policy/continuous_aggregate_api.c

```diff
@@ -111,13 +111,20 @@ get_time_from_config(const Dimension *dim, const Jsonb *config, const char *json
 }
 
 int64
-policy_refresh_cagg_get_refresh_start(const Dimension *dim, const Jsonb *config, bool *start_isnull)
+policy_refresh_cagg_get_refresh_start(const ContinuousAgg *cagg, const Dimension *dim,
+									  const Jsonb *config, bool *start_isnull)
 {
 	int64 res = get_time_from_config(dim, config, POL_REFRESH_CONF_KEY_START_OFFSET, start_isnull);
 
 	/* interpret NULL as min value for that type */
 	if (*start_isnull)
-		return ts_time_get_min(ts_dimension_get_partition_type(dim));
+	{
+		Oid type = ts_dimension_get_partition_type(dim);
+
+		return ts_continuous_agg_bucket_width_variable(cagg) ? ts_time_get_nobegin_or_min(type) :
```

Comment:
Those functions to get min/max for a given type are always confusing... in this case we can use `ts_time_get_nobegin` because integer time dimensions don't allow variable bucket sizes or am I missing something?

Reply:
On the other hand we can encapsulate this into a helper function, something like:
```C
bool
ts_continuous_agg_time_get_min(ContinuousAgg *cagg)
{
    return ts_continuous_agg_bucket_width_variable(cagg) ? ts_time_get_nobegin_or_min(cagg->partition_type) : ts_time_get_min(cagg->partition_type);
}
```

Reply:
But this can be part of a following PR ... I can do it with a pleasure!